mobile_verifier/src/reward_shares.rs

```diff
@@ -203,52 +204,147 @@ pub fn dc_to_mobile_bones(dc_amount: Decimal, mobile_bone_price: Decimal) -> Dec
         .round_dp_with_strategy(DEFAULT_PREC, RoundingStrategy::ToPositiveInfinity)
 }
 
-#[derive(Default)]
-pub struct RadioShares {
-    radio_shares: HashMap<Option<String>, Decimal>,
+#[derive(Debug)]
+struct RadioPoints {
+    heartbeat_multiplier: Decimal,
+    coverage_object: Uuid,
+    seniority: DateTime<Utc>,
+    points: Decimal,
 }
 
-impl RadioShares {
-    fn total_shares(&self) -> Decimal {
-        self.radio_shares
-            .values()
-            .fold(Decimal::ZERO, |sum, amount| sum + amount)
+impl RadioPoints {
+    fn new(heartbeat_multiplier: Decimal, coverage_object: Uuid, seniority: DateTime<Utc>) -> Self {
+        Self {
+            heartbeat_multiplier,
+            seniority,
+            coverage_object,
+            points: Decimal::ZERO,
+        }
+    }
+
+    fn points(&self) -> Decimal {
+        (self.heartbeat_multiplier * self.points).max(Decimal::ZERO)
     }
 }
 
-#[derive(Default)]
-pub struct PocShares {
-    pub hotspot_shares: HashMap<PublicKeyBinary, RadioShares>,
+#[derive(Debug, Default)]
+struct HotspotPoints {
+    /// Points are multiplied by the multiplier to get shares.
+    /// Multiplier should never be zero.
+    speedtest_multiplier: Decimal,
+    radio_points: HashMap<Option<String>, RadioPoints>,
 }
 
-impl PocShares {
-    pub async fn aggregate(
-        heartbeat_rewards: impl Stream<Item = Result<HeartbeatReward, sqlx::Error>>,
-        speedtest_averages: &SpeedtestAverages,
-    ) -> anyhow::Result<Self> {
-        let mut poc_shares = Self::default();
-        let mut heartbeat_rewards = std::pin::pin!(heartbeat_rewards);
-        while let Some(heartbeat_reward) = heartbeat_rewards.next().await.transpose()? {
-            let speedmultiplier = speedtest_averages
-                .get_average(&heartbeat_reward.hotspot_key)
+impl HotspotPoints {
+    pub fn new(speedtest_multiplier: Decimal) -> Self {
+        Self {
+            speedtest_multiplier,
```

Comment:
maybe return a result here and check that the passed-in multiplier is greater than Decimal::ZERO?

Reply:
I don't think that's particularly useful? It wouldn't be very relevant to this PR either